### PR TITLE
Add userOrSource to GenericArguments

### PR DESCRIPTION
**Pull request by pie-flavor**
userOrSource behaves like playerOrSource

### CODE REVIEW

File: src/main/java/org/spongepowered/api/command/args/GenericArguments.java [hunk 1]
```diff
@@ -949,7 +962,14 @@ protected Object parseValue(CommandSource source, CommandArgs args) throws Argum
                 return this.possiblePlayer.parseValue(source, args);
             } catch (ArgumentParseException ex) {
                 args.setState(state);
-                return super.parseValue(source, args);
+                try {
+                    return super.parseValue(source, args);
+                } catch (ArgumentParseException ex2) {
+                    if (returnSource && source instanceof User)
```
Comment (kashike) [inline src/main/java/org/spongepowered/api/command/args/GenericArguments.java]: We require brackets on `if` statements.

File: src/main/java/org/spongepowered/api/command/args/GenericArguments.java [hunk 2]
```diff
@@ -949,7 +962,14 @@ protected Object parseValue(CommandSource source, CommandArgs args) throws Argum
...
+                        return source;
+                    else
```
Comment (kashike) [inline src/main/java/org/spongepowered/api/command/args/GenericArguments.java]: We require brackets on `else` statements.

File: src/main/java/org/spongepowered/api/command/args/GenericArguments.java [hunk 3]
```diff
@@ -125,7 +125,19 @@ public static CommandElement player(Text key) {
      * @return the argument
      */
     public static CommandElement user(Text key) {
-        return new UserCommandElement(key);
+        return new UserCommandElement(key, false);
+    }
+    /**
```
Comment (kenzierocks) [inline src/main/java/org/spongepowered/api/command/args/GenericArguments.java]: Should be a newline between methods.

File: src/main/java/org/spongepowered/api/command/args/GenericArguments.java [hunk 4]
```diff
@@ -935,10 +947,11 @@ public Text getUsage(CommandSource src) {
 
     private static class UserCommandElement extends PatternMatchingCommandElement {
         private final PlayerCommandElement possiblePlayer;
-
-        protected UserCommandElement(@Nullable Text key) {
+        private final boolean returnSource;
+        protected UserCommandElement(@Nullable Text key, boolean returnSource) {
```
Comment (kenzierocks) [inline src/main/java/org/spongepowered/api/command/args/GenericArguments.java]: Newline between field and constructor.

File: src/main/java/org/spongepowered/api/command/args/GenericArguments.java [hunk 5]
```diff
@@ -949,7 +962,15 @@ protected Object parseValue(CommandSource source, CommandArgs args) throws Argum
                 return this.possiblePlayer.parseValue(source, args);
             } catch (ArgumentParseException ex) {
                 args.setState(state);
-                return super.parseValue(source, args);
+                try {
+                    return super.parseValue(source, args);
+                } catch (ArgumentParseException ex2) {
+                    if (returnSource && source instanceof User) {
```
Comment (kashike) [inline src/main/java/org/spongepowered/api/command/args/GenericArguments.java]: `this.returnSource`

File: src/main/java/org/spongepowered/api/command/args/GenericArguments.java [hunk 6]
```diff
@@ -125,7 +125,20 @@ public static CommandElement player(Text key) {
...
     public static CommandElement user(Text key) {
-        return new UserCommandElement(key);
+        return new UserCommandElement(key, false);
+    }
+    
+    /**
+     * Expect an argument to return a player who has been online at some point, as a {@link User},
+     * or if nothing matches and the source is a {@link User}, give the user. If nothing matches and the source is not a
+     * {@link User}, throw an exception.
+     * Gives value of type {@link User}
```
Comment (gabizou) [inline src/main/java/org/spongepowered/api/command/args/GenericArguments.java]: Fix the formatting of this javadoc, it's all over the place.

File: src/main/java/org/spongepowered/api/command/args/GenericArguments.java [hunk 7]
```diff
@@ -125,7 +125,19 @@ public static CommandElement player(Text key) {
      * @return the argument
      */
     public static CommandElement user(Text key) {
-        return new UserCommandElement(key);
+        return new UserCommandElement(key, false);
+    }
+    
+    /**
+     * Expect an argument to represent a player who has been online at some point, as a {@link User},
```
Comment (kashike) [inline src/main/java/org/spongepowered/api/command/args/GenericArguments.java]: Try to limit javadoc lines to ~85 characters, please

**Pull request closed without merging**